api/envoy/api/v2/route/route.proto

```diff
@@ -771,6 +784,19 @@ message RouteAction {
   // occurs.
   google.protobuf.Duration idle_timeout = 24;
 
+  message RegexRewrite {
+    // Represents a regular expression that the path must match in order for the substitution
+    // to be applied. The regular expression may include capture groups to be interpolated into the
+    // substitution.
+    string pattern = 1;
```

Comment:
I think you need to integrate with https://github.com/envoyproxy/envoy/blob/master/api/envoy/type/matcher/regex.proto to support compatibility with the full range of regex engines that we might support.